File: MiniDOMBase/DMD/Sistema/LoginHistory/CLoginHistoryCursor.vb
Imports minidom.Databases

Partial Public Class Sistema

    ''' <summary>
    ''' Cursore sulla tabella dei log di accesso
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CLoginHistoryCursor
        Inherits DBObjectCursorBase(Of CLoginHistory)

        Private m_UserID As CCursorField(Of Integer)
        Private m_LoginTime As CCursorField(Of Date)
        Private m_LogoutTime As CCursorField(Of Date)
        Private m_LogoutMethod As CCursorField(Of LogOutMethods)
        Private m_RemoteIP As CCursorFieldObj(Of String)
        Private m_RemotePort As CCursorField(Of Integer)
        Private m_UserAgent As CCursorFieldObj(Of String)

        Public Sub New()
            Me.m_UserID = New CCursorField(Of Integer)("User")
            Me.m_LoginTime = New CCursorField(Of Date)("LogInTime")
            Me.m_LogoutTime = New CCursorField(Of Date)("LogOutTime")
            Me.m_LogoutMethod = New CCursorField(Of LogOutMethods)("LogoutMethod")
            Me.m_RemoteIP = New CCursorFieldObj(Of String)("RemoteIP")
            Me.m_RemotePort = New CCursorField(Of Integer)("RemotePort")
            Me.m_UserAgent = New CCursorFieldObj(Of String)("UserAgent")
        End Sub



        Public ReadOnly Property UserID As CCursorField(Of Integer)
            Get
                Return Me.m_UserID
            End Get
        End Property

        Public ReadOnly Property LoginTime As CCursorField(Of Date)
            Get
                Return Me.m_LoginTime
            End Get
        End Property

        Public ReadOnly Property LogoutTime As CCursorField(Of Date)
            Get
                Return Me.m_LogoutTime
            End Get
        End Property

        Public ReadOnly Property LogoutMethod As CCursorField(Of LogOutMethods)
            Get
                Return Me.m_LogoutMethod
            End Get
        End Property

        Public ReadOnly Property RemoteIP As CCursorFieldObj(Of String)
            Get
                Return Me.m_RemoteIP
            End Get
        End Property

        Public ReadOnly Property RemotePort As CCursorField(Of Integer)
            Get
                Return Me.m_RemotePort
            End Get
        End Property

        Public ReadOnly Property UserAgent As CCursorFieldObj(Of String)
            Get
                Return Me.m_UserAgent
            End Get
        End Property

        Public Overrides Function GetTableName() As String
            Return "tbl_LoginHistory"
        End Function

        'Protected Overrides Function GetCursorFields() As CCursorFieldsCollection
        '    Dim col As CCursorFieldsCollection
        '    col = MyBase.GetCursorFields
        '    col.Add(m_UserID)
        '    col.Add(m_LoginTime)
        '    col.Add(m_LogoutTime)
        '    col.Add(m_LogoutMethod)
        '    col.Add(m_RemoteIP)
        '    col.Add(m_RemotePort)
        '    col.Add(m_UserAgent)
        '    Return col
        'End Function

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New CLoginHistory
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return LOGConn
        End Function
    End Class

End Class

File: MiniDOMBase/DMD/Sistema/LoginHistory/CLoginHistoryCursor.cs
using System;
using System.Collections;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using static minidom.Sistema;
using static minidom.Anagrafica;
using minidom.repositories;
using System.Collections.Generic;


namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Cursore sulla tabella dei log di accesso
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CLoginHistoryCursor 
            : minidom.Databases.DBObjectCursorBase<CLoginHistory>
        {
            private DBCursorField<int> m_UserID;
            private DBCursorField<DateTime> m_LoginTime;
            private DBCursorField<DateTime> m_LogoutTime;
            private DBCursorField<LogOutMethods> m_LogoutMethod;
            private DBCursorStringField m_RemoteIP;
            private DBCursorField<int> m_RemotePort;
            private DBCursorStringField m_UserAgent;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CLoginHistoryCursor()
            {
                m_UserID = new DBCursorField<int>("User");
                m_LoginTime = new DBCursorField<DateTime>("LogInTime");
                m_LogoutTime = new DBCursorField<DateTime>("LogOutTime");
                m_LogoutMethod = new DBCursorField<LogOutMethods>("LogoutMethod");
                m_RemoteIP = new DBCursorStringField("RemoteIP");
                m_RemotePort = new DBCursorField<int>("RemotePort");
                m_UserAgent = new DBCursorStringField("UserAgent");
            }

            /// <summary>
            /// UserID
            /// </summary>
            public DBCursorField<int> UserID
            {
                get
                {
                    return m_UserID;
                }
            }

            /// <summary>
            /// LoginTime
            /// </summary>
            public DBCursorField<DateTime> LoginTime
            {
                get
                {
                    return m_LoginTime;
                }
            }

            /// <summary>
            /// LogoutTime
            /// </summary>
            public DBCursorField<DateTime> LogoutTime
            {
                get
                {
                    return m_LogoutTime;
                }
            }

            /// <summary>
            /// LogoutMethod
            /// </summary>
            public DBCursorField<LogOutMethods> LogoutMethod
            {
                get
                {
                    return m_LogoutMethod;
                }
            }

            /// <summary>
            /// RemoteIP
            /// </summary>
            public DBCursorStringField RemoteIP
            {
                get
                {
                    return m_RemoteIP;
                }
            }

            /// <summary>
            /// RemotePort
            /// </summary>
            public DBCursorField<int> RemotePort
            {
                get
                {
                    return m_RemotePort;
                }
            }

            /// <summary>
            /// UserAgent
            /// </summary>
            public DBCursorStringField UserAgent
            {
                get
                {
                    return m_UserAgent;
                }
            }

            //public override string GetTableName()
            //{
            //    return "tbl_LoginHistory";
            //}

            // Protected Overrides Function GetCursorFields() As CCursorFieldsCollection
            // Dim col As CCursorFieldsCollection
            // col = MyBase.GetCursorFields
            // col.Add(m_UserID)
            // col.Add(m_LoginTime)
            // col.Add(m_LogoutTime)
            // col.Add(m_LogoutMethod)
            // col.Add(m_RemoteIP)
            // col.Add(m_RemotePort)
            // col.Add(m_UserAgent)
            // Return col
            // End Function

            /// <summary>
            /// Istanzia un nuovo oggetto
            /// </summary>
            /// <param name="dbRis"></param>
            /// <returns></returns>
            public override CLoginHistory InstantiateNewT(DBReader dbRis)
            {
                return new CLoginHistory();
            }

            /// <summary>
            /// Respository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Sistema.Users.LoginHistories;
            }

            //protected internal override CDBConnection GetConnection()
            //{
            //    return LOGConn;
            //}
        }
    }
}